File: src/main/scala/com/bjsxt/sparkscalacode/SparkSecondSort.scala
package com.bjsxt.sparkscalacode

import org.apache.spark.{SparkConf, SparkContext}

object SparkSecondSort {
  def main(args: Array[String]): Unit = {
    val sc = new SparkContext(new SparkConf().setMaster("local").setAppName("test"))

    val lines = sc.textFile("./data/secondSort.txt")

    lines.map(line=>(MySort2(line.split(" ")(0).toInt,line.split(" ")(1).toInt),line)).sortByKey(false).foreach(s=>println(s._2))
  }
}
case class MySort2(x1:Int,x2:Int) extends Ordered[MySort2] {
  override def compare(that: MySort2): Int = {
    if (this.x1 == that.x1){
      this.x2 - that.x2
    }
    this.x1 - that.x1
  }
}

File: src/main/java/com/bjsxt/sparkjavacode/SparkSecondSort.java
package com.bjsxt.sparkjavacode;

import org.apache.spark.SparkConf;
import org.apache.spark.api.java.JavaRDD;
import org.apache.spark.api.java.JavaSparkContext;
import org.apache.spark.api.java.function.PairFunction;
import org.apache.spark.api.java.function.VoidFunction;
import scala.Tuple2;

public class SparkSecondSort {
    public static void main(String[] args) {
        JavaSparkContext sc = new JavaSparkContext(new SparkConf().setMaster("local").setAppName("test"));

        JavaRDD<String> lines = sc.textFile("./data/secondSort.txt");

        lines.mapToPair(s->new Tuple2<MySort,String>(new MySort(Integer.valueOf(s.split(" ")[0]),Integer.valueOf(s.split(" ")[1])),s)).
                sortByKey(false).foreach(s-> System.out.println(s._2));


        lines.mapToPair(new PairFunction<String, MySort, String>() {
            @Override
            public Tuple2<MySort, String> call(String s) throws Exception {
                return new Tuple2<>(new MySort(Integer.valueOf(s.split(" ")[0]),Integer.valueOf(s.split(" ")[1])),s);
            }
        }).sortByKey(false).foreach(new VoidFunction<Tuple2<MySort, String>>() {
            @Override
            public void call(Tuple2<MySort, String> mySortStringTuple2) throws Exception {
                System.out.println(mySortStringTuple2._2);
            }
        });
    }
}
